web/src/components/core/Section.jsx

```diff
@@ -23,148 +23,130 @@
 
 import React from "react";
 import { Link } from "react-router-dom";
-import {
-  Button,
-  Text,
-  TextContent,
-  TextVariants,
-  Tooltip
-} from "@patternfly/react-core";
-
+import { Text, TextVariants } from "@patternfly/react-core";
 import { Icon } from '~/components/layout';
 import { ValidationErrors } from "~/components/core";
 
 /**
- * Helper method for rendering section icon
+ * Internal component for rendering the section icon
  *
- * @param {string} name
- * @param {number} [size=32]
+ * @param {object} props
+ * @param {string} [props.name] - the name of the icon
+ * @param {number} [props.size=32] - the icon size
  *
- * @return {React.ReactNode}
+ * @return {React.ReactElement}
  */
-const renderIcon = (name, size = 32) => {
+const SectionIcon = ({ name, size = 32 }) => {
   if (!name) return null;
 
   return <Icon name={name} size={size} aria-hidden />;
 };
 
+/**
+ * Internal component for rendering the section title
+ *
+ *  NOTE: a section can do either, navigate to a page following given path or
+ *  open a dialog triggering the openDialog callback but not both. Thus, if path
+ *  is given onDialogCallback will be completely ignored. page or open a dialog,
+ *  but not both.
+ *
+ * @param {object} props
+ * @param {string} props.text - the title for the section
+ * @param {string} props.path - the path where the section links to. If present, props.openDialog is ignored
+ * @param {React.MouseEventHandler|undefined} [props.openDialog] - callback to be triggered when user clicks on the title, used for opening a dialog.
+ *
+ * @return {JSX.Element}
+ */
+const SectionTitle = ({ text, path, openDialog }) => {
+  let title = <>{text}</>;
+
+  if (path && path !== "") {
+    title = <Link to={path}>{text}</Link>;
+  } else if (typeof openDialog === "function") {
+    // NOTE: using a native button here on purpose
+    title = <button onClick={openDialog}>{text}</button>;
+  }
+
+  return (
+    <Text component={TextVariants.h2}>
+      {title}
+    </Text>
+  );
+};
+
+/**
+ * Internal component for wrapping and rendering the section content
+ *
+ * @param {object} props
+ * @param {React.ReactElement|React.ReactElement[]} props.children - the content to be wrapped
+ * @return {JSX.Element}
+ */
+const SectionContent = ({ children }) => {
+  return (
+    <div className="stack content">
+      {children}
+    </div>
+  );
+};
+
 /**
  *
  * Displays an installation section
  * @component
  *
- * @example <caption>Simple usage</caption>
- *   <Section title="Users" icon={UsersIcon}>
- *     <UserSectionContent />
- *   </Section>
+ *  NOTE: a section can do either, navigate to a page following given path or
```

Comment:
DRY :-) If the behavior changes, you need to remember to change both comments.

Reply:
You're right. And I knew it... but  I was looking for more chances to have everyone warned. Anyway, I'm going to follow your advice and remove the comment from the internal component. Thanks!

Reply:
>  I'm going to follow your advice

Which was helpful because both comments had leftovers :man_facepalming: